### PR TITLE
Fix logs not getting created and uploading to the same s3 location by both Prod and Dev.

### LINKED ISSUE #774: [S3] Dev and Prod both persist/refer logs in same location.
**Description:**
Currently both Dev and Prod test flows refer/persist logs in same S3 location. 
Ex: For WSO2-IS-5.4.1-LTS, the logs are stored in following location by both dev and prod:
Amazon S3 >jenkins-testrun-artifacts/artifacts/jobs/wso2is-5.4.1-LTS/workspace/

**Suggested Labels:**
Critical/ Bug/ Highest Priority

**Affected Product Version:**
m29
**OS, DB, other environment details and versions:**    
N/A

### PR DESCRIPTION
**Pull request by pasindujw**
**Purpose**
Fixes #774 #782

**Goals**
Fix bugs related to TestGrid logging.

**Approach**
Added correct log4j, slf4j dependencies to logging package.
Parameterized S3 bucket so that each dev, prod instance can point to different buckets.
 

**Security checks**
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: common/src/main/java/org/wso2/testgrid/common/config/ConfigurationContext.java [hunk 1]
```diff
@@ -105,6 +105,11 @@ public static String getProperty(Enum property) {
          */
         AWS_REGION_NAME("AWS_REGION_NAME"),
 
+        /**
+         * AWS Bucket use to upload TestGrid artifacts
```
Comment (harshanL) [inline common/src/main/java/org/wso2/testgrid/common/config/ConfigurationContext.java]: Should be "used to"

File: web/src/main/java/org/wso2/testgrid/web/api/ProductService.java [hunk 2]
```diff
@@ -203,7 +203,7 @@ public Response isProductReportExist(
                 String bucketKey = Paths.get(Constants.AWS_BUCKET_ARTIFACT_DIR, productName, fileName).toString();
                 ArtifactReadable artifactReadable = new AWSArtifactReader(ConfigurationContext.
                         getProperty(ConfigurationContext.ConfigurationProperties.AWS_REGION_NAME),
-                        Constants.AWS_BUCKET_NAME);
+                        ConfigurationContext.getProperty(ConfigurationContext.ConfigurationProperties.AWS_BUCKET_NAME));
                 if (artifactReadable.isExistArtifact(bucketKey)) {
```
Comment (harshanL) [inline web/src/main/java/org/wso2/testgrid/web/api/ProductService.java]: We should rename the method "isExistArtifact" to something like "isArtifactExist" or "isExist". WDYT?

File: jenkins/pipelines/test-jobs/wso2apim-2.1.0-LTS/Jenkinsfile [hunk 3]
```diff
@@ -1,3 +1,9 @@
+properties = null
+def loadProperties() {
+    node {
+        properties = readProperties file: "${TESTGRID_HOME}/config.properties"
```
Comment (kasunbg) [inline jenkins/pipelines/test-jobs/wso2apim-2.1.0-LTS/Jenkinsfile]: Shall we set default bucket name if the properties does not contain the property?

Comment (pasindujw) [reply jenkins/pipelines/test-jobs/wso2apim-2.1.0-LTS/Jenkinsfile]: +1 Added.

**Pull request merged**